app/controllers/books_controller.rb

```diff
@@ -4,6 +4,23 @@ class BooksController < ApplicationController
   def index
   end
 
+  def show
+    book = Book.find(params[:id])
+    @book = {
+      :book => book,
+      :image => (book.large_image_link ? book.large_image_link : Book.no_image_link),
+      :link => (book.amazon_affiliate_link ? book.amazon_affiliate_link : "#"),
+      :new_bookstore => (book.bookstore_new_price ? "$" + sprintf('%.2f', book.bookstore_new_price) : "N/A"),
+      :used_bookstore => (book.bookstore_used_price ? "$" + sprintf('%.2f', book.bookstore_used_price) : "N/A"),
+      :new_official_amazon => (book.amazon_official_new_price ? "$" + sprintf('%.2f', book.amazon_official_new_price) : "N/A"),
+      :used_official_amazon => (book.amazon_official_used_price ? "$" + sprintf('%.2f', book.amazon_official_used_price) : "N/A"),
+      :new_merchant_amazon => (book.amazon_merchant_new_price ? "$" + sprintf('%.2f', book.amazon_merchant_new_price) : "N/A"),
```

Comment:
you should technically actually add this as utility method to book.rb, something like def format